### PR TITLE
Fix API Key calls

### PR DESCRIPTION
**Pull request by rromanowski-figure**
## Description

Address issue with looking up secrets for consumers

closes: https://app.rollbar.com/a/figure/fix/item/p8e-cee-api/6021

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/p8e-cee-api/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Unit Test Results` in the comment section below once CI passes

### CODE REVIEW

File: models/src/main/kotlin/io/provenance/api/models/entity/EntityID.kt [hunk 1]
```diff
@@ -15,10 +15,10 @@ sealed interface Entity { val id: String }
 data class KongConsumer(
     val entityId: String,
     val username: String?,
-    val customId: String,
+    val customId: String?,
 ) : Entity {
     override val id: String
-        get() = entityId
+        get() = customId ?: username ?: entityId
```
Comment (rromanowski-figure) [inline models/src/main/kotlin/io/provenance/api/models/entity/EntityID.kt:21]: core update

File: service/src/main/kotlin/io/provenance/api/domain/usecase/cee/common/ContractUtilities.kt [hunk 2]
```diff
@@ -69,7 +69,7 @@ class ContractUtilities(
                 participantsMap,
                 scopeToUse,
                 config.contract.scopeSpecificationName,
-                audiences.map { it.encryptionKey.toJavaPublicKey() }.toSet()
+                audiences.map { audience -> audience.encryptionKey.toJavaPublicKey() }.toSet()
```
Comment (rromanowski-figure) [inline service/src/main/kotlin/io/provenance/api/domain/usecase/cee/common/ContractUtilities.kt:72]: shadowed `it`

File: service/src/main/kotlin/io/provenance/api/domain/usecase/cee/approve/ApproveContractBatchExecution.kt [hunk 3]
```diff
@@ -66,7 +66,7 @@ class ApproveContractBatchExecution(
                     }
                 }.fold(
                     onSuccess = {
-                        log.info("Successfully processed batch $index of ${chunked.size}")
+                        log.debug { "Successfully processed batch $index of ${chunked.size}" }
```
Comment (rromanowski-figure) [inline service/src/main/kotlin/io/provenance/api/domain/usecase/cee/approve/ApproveContractBatchExecution.kt:69]: limiting what is always logged

File: service/src/main/kotlin/io/provenance/api/frameworks/web/misc/ServerRequestExt.kt [hunk 4]
```diff
@@ -37,9 +37,7 @@ private fun ServerRequest.getConsumerOrNull(): KongConsumer? {
         KongConsumer(
             entityId = id,
             username = headers().firstHeader("x-consumer-username"),
-            customId = requireNotNull(headers().firstHeader("x-consumer-custom-id")) {
-                "Custom ID for consumer (x-consumer-custom-id) missing"
-            },
+            customId = headers().firstHeader("x-consumer-custom-id"),
```
Comment (rromanowski-figure) [inline service/src/main/kotlin/io/provenance/api/frameworks/web/misc/ServerRequestExt.kt:40]: allow nullable since some consumers were setup without them

**Review by rpatel-figure — APPROVED** [review]
Thanks for the cleanup!

**Review by rpatel-figure — APPROVED** [review]

**Pull request merged**